File: src/main/java/models/tournament/internal/Tournament.java
package models.tournament.internal;

import lombok.Builder;
import lombok.NonNull;
import lombok.Value;
import lombok.extern.jackson.Jacksonized;
import models.player.internal.Rating;

import java.util.List;
import java.util.UUID;

@Value
@Builder
@Jacksonized
public class Tournament {

    UUID id;

    @NonNull
    String name;

    String description;

    @NonNull
    String city;

    String prizes;

    @NonNull
    Long startDate;

    @NonNull
    Long endDate;

    @NonNull
    List<UUID> organizerIds;

    @NonNull
    TournamentStatus status;

    @NonNull
    Rating minRating;

    @NonNull
    Rating maxRating;

    Long createdAt;

    Long updatedAt;

}

File: frontend/src/model/tournament/Tournament.ts
export type TournamentStatus = "pending" | "active" | "completed";

export const STATUSES: TournamentStatus[] = ["pending", "active", "completed"];

export interface Tournament {
    id: string;
    name: string;
    description?: string;
    city: string;
    prizes?: string;
    startDate: number;
    endDate: number;
    organizerIds: string[];
    status: TournamentStatus;
    minRating: number;
    maxRating: number;
}